### PR TITLE
[docs] add help post about showing toasts in RN

### PR DESCRIPTION
**Pull request by FiberJW**
# Why

React Native devs often want to use Toasts in their apps, so this short doc will help guide them in the right direction.

# How

- added docs page under ui-programming
- covers RN ToastAndroid
- covers https://github.com/magicismight/react-native-root-toast
- ImageOptim'ed the screenshot

# Test Plan

`yarn dev` in docs and visit http://localhost:3002/ui-programming/react-native-toast/ to read the page.

### CODE REVIEW

File: docs/pages/ui-programming/react-native-toast.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+<ImageSpotlight style={{maxWidth: 360}} alt="Pixel 3a showing toast message in an Expo app" src="/static/images/ToastAndroid.png" />
+
+There are many other ways to configure your toast position, duration, and gravity options. Read the
+[React Native `ToastAndroid`](https://reactnative.dev/docs/toastandroid) docs to learn more.
+
+## `react-native-root-toast`
+
+Since iOS doesn't have a built-in toast feature, React Native developers have to implement their own
+cross-platform toast libraries.
+[`react-native-root-toast`](https://github.com/magicismight/react-native-root-toast) is one such
```
Comment (tcdavis) [inline docs/pages/ui-programming/react-native-toast.md:68]: Would be nice to talk to why you are recommending this module over the other community solutions

File: docs/pages/ui-programming/react-native-toast.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+}, 500);
+```
+
+`react-native-root-toast` also has a component API if you want to manage your toasts declaratively.
+
+```jsx
+<Toast visible={this.state.visible}>Thanks for subscribing!</Toast>
+```
+
+This library also has a lot of options for customizing appearance and behavior, so check out [their
```
Comment (tcdavis) [inline docs/pages/ui-programming/react-native-toast.md:110]:
```suggestion
This library has many options for [customizing the appearance and behavior](https://github.com/crazycodeboy/react-native-easy-toast#api) of your toast.
```

File: docs/pages/ui-programming/react-native-toast.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
+---
+title: How to display a toast in React Native
+sidebar_title: Displaying toast messages
+---
+
+import ImageSpotlight from '~/components/plugins/ImageSpotlight';
+
+## What is a toast?
+
+According to the [Android Developers
```
Comment (tcdavis) [inline docs/pages/ui-programming/react-native-toast.md:10]: Something about why you might use a toast: Toasts are the standard technique to notify your users about something without interrupting what they are doing.

File: docs/pages/ui-programming/react-native-toast.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+Documentation](https://developer.android.com/guide/topics/ui/notifiers/toasts): "A toast provides
+simple feedback about an operation in a small popup. It only fills the amount of space required for
+the message and the current activity remains visible and interactive. Toasts automatically disappear
+after a timeout".
+
+To present a toast, we recommend two solutions: an API from React Native (RN) itself and a library
+maintained by the RN community.
+
+## `ToastAndroid`
```
Comment (tcdavis) [inline docs/pages/ui-programming/react-native-toast.md:19]: Android-only: `ToastAndroid`

File: docs/pages/ui-programming/react-native-toast.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+```
+
+The code above results in this on a Pixel 3a:
+
+<ImageSpotlight style={{maxWidth: 360}} alt="Pixel 3a showing toast message in an Expo app" src="/static/images/ToastAndroid.png" />
+
+There are many other ways to configure your toast position, duration, and gravity options. Read the
+[React Native `ToastAndroid`](https://reactnative.dev/docs/toastandroid) docs to learn more.
+
+## `react-native-root-toast`
```
Comment (tcdavis) [inline docs/pages/ui-programming/react-native-toast.md:64]: Cross-platform: `react-native-root-toast`

**Review by tcdavis — CHANGES_REQUESTED** [review]
A couple changes, but looking good

File: docs/pages/ui-programming/react-native-toast.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
+---
+title: How to display a toast in React Native
```
Comment (ide) [inline docs/pages/ui-programming/react-native-toast.md:2]:
```suggestion
title: How to display a popup toast
```

something like this fits in better with the docs and also hints at what a toast is. also would name this file just toast.md.

File: docs/pages/ui-programming/react-native-toast.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+Toasts are the standard technique in mobile development for notifying your users about something without interrupting what they are doing.
+
+According to the [Android Developers
+Documentation](https://developer.android.com/guide/topics/ui/notifiers/toasts): "A toast provides
+simple feedback about an operation in a small popup. It only fills the amount of space required for
+the message and the current activity remains visible and interactive. Toasts automatically disappear
+after a timeout".
+
+To present a toast, we recommend two solutions: an API from React Native itself and a library
```
Comment (ide) [inline docs/pages/ui-programming/react-native-toast.md:18]:
```suggestion
To present a toast, we recommend two solutions: an API from the `react-native` package and a library
```
this wording communicates better that it's not really a core component and more an arbitrary component

File: docs/pages/ui-programming/react-native-toast.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+[`react-native-root-toast`](https://github.com/magicismight/react-native-root-toast) is one such
+solution that the developer has shared with the React Native community.
+
+We recommend this solution because it's one of the most used and maintained open-source libraries that
+work on iOS and Android without the need for native code. It also provides a lot of customization
+options, which means that you will be able to match the design of your toasts to the rest of your app.
+
+### Usage
+
+To use `react-native-root-toast`, you have to [install the module](https://github.com/magicismight/react-native-root-toast) from NPM with `npm install react-native-root-toast`.
```
Comment (ide) [inline docs/pages/ui-programming/react-native-toast.md:79]: npm is lowercase
```suggestion
To use `react-native-root-toast`, you have to [install the module](https://github.com/magicismight/react-native-root-toast) from npm with `npm install react-native-root-toast`.
```

**Review by ide — CHANGES_REQUESTED** [review]

File: docs/constants/navigation.js [hunk 9]
```diff
@@ -195,6 +195,7 @@ const sections = [
       "Setting a component's background image",
       'Implementing a checkbox for Expo and React Native apps',
       'Stacking overlapping views with zIndex in Expo and React Native apps',
+      'How to display a toast in React Native',
```
Comment (ide) [inline docs/constants/navigation.js:198]:
```suggestion
      'How to display a popup toast',
```

**Review by ide — APPROVED** [review]

**Pull request merged**